effect/shared/src/main/scala/com/monovore/decline/effect/CommandIOApp.scala

```diff
@@ -1,21 +1,26 @@
 package com.monovore.decline.effect
 
-import cats.effect.{IO, IOApp, ExitCode}
+import cats.effect.implicits._
+import cats.effect.{Effect, ExitCode, IO, IOApp}
 import cats.implicits._
-
 import com.monovore.decline._
 
-abstract class CommandIOApp(
+import scala.language.higherKinds
+
+abstract class CommandIOApp[F[_]: Effect](
```

Comment:
actually, I guess mima must not be running, the `: Effect` will change the signature of this class to require the `Effect[F]` so I think this breaks binary compatibility.

I think it is better not to break binary compatibility personally. What about adding:
```scala
abstract class CommandEffectApp[F[_]: Effect]...
```
then make:
```scala
abstract class CommandIOApp(...) extends CommandEffectApp[IO](name, header,...)
```
That way, I don't think we break binary compatibility, but also ZIO users can use `CommandEffectApp[ZIO[X, ?]]` or whatever...

Reply:
See my comment below, if you agree I would keep this as a backup solution in case we don't want to go with what @bkirwi proposed